go-controller/pkg/ovn/pods.go

```diff
@@ -28,10 +28,14 @@ func (oc *Controller) syncPods(pods []interface{}) {
 			klog.Errorf("Spurious object in syncPods: %v", podInterface)
 			continue
 		}
-		_, err := util.UnmarshalPodAnnotation(pod.Annotations)
+		annotations, err := util.UnmarshalPodAnnotation(pod.Annotations)
 		if podScheduled(pod) && podWantsNetwork(pod) && err == nil {
 			logicalPort := podLogicalPortName(pod)
 			expectedLogicalPorts[logicalPort] = true
+			podIP := annotations.IPs[0].IP
+			if err = oc.lsCache.AllocateIP(pod.Spec.NodeName, &podIP); err != nil {
```

Comment:
I'm not sure this is correct; the kubernetes API is the source of truth.

If we can't allocate this IP, then we have an internal consistency problem and need to stop and figure out what's going on. But, always, the API is true and we need to adjust our state to that.

Reply:
What do you suggest we do in this case beyond logging error?